### PR TITLE
Post Release-1.2 fixes

### PR DESCRIPTION
**Pull request by JackReevies**
Contains PRs made after code freeze for 1.2

Tickets:

https://tools.hmcts.net/jira/browse/RET-3675
https://tools.hmcts.net/jira/browse/RET-3670
https://tools.hmcts.net/jira/browse/RET-3907
https://tools.hmcts.net/jira/browse/RET-3397

.... Feel free to edit this if I've missed anything @Itay-Kirsh

### CODE REVIEW

**Review by Itay-Kirsh — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/DocumentManagementService.java [hunk 1]
```diff
@@ -205,7 +205,7 @@ public UploadedDocumentType addDocumentToDocumentField(DocumentInfo documentInfo
      */
     public String displayDocNameTypeSizeLink(UploadedDocumentType document, String authToken) {
         if (document == null) {
-            return "";
+            return null;
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/DocumentManagementService.java:208]: assuming this null is handled correctly when returned

File: src/main/resources/application-cftlib.properties [hunk 2]
```diff
@@ -7,33 +7,33 @@ spring.flyway.locations=classpath:db/migration,classpath:db/dev
...
-rd_professional.api.url=http://localhost:4507
+rd_professional-api-url=http://localhost:4507
 feature.secure-doc-store.enabled=${SECURE_DOC_STORE_FEATURE:false}
 
 uk.gov.notify.api.key=${GOV_NOTIFY_API_KEY:someApiKey}
-referral.template.id=${REFERRAL_TEMPLATE_ID:12345}
-et1Serving.template.id=${ET1SERVING_TEMPLATE_ID:12345}
-et1Serving.respondent.template.id=${ET1SERVING_RESPONDENT_TEMPLATE_ID:12345}
-rejected.template.id=${REJECTED_TEMPLATE_ID:12345}
+template.referral=${REFERRAL_TEMPLATE_ID:12345}
```
Comment (hsjhita) [inline src/main/resources/application-cftlib.properties:27]: if all the base templates exist in application.yaml, maybe we don't need these version in cftlib?

Comment (hsjhita) [reply src/main/resources/application-cftlib.properties:27]: especially given that some are just dummy values

Comment (hsjhita) [reply src/main/resources/application-cftlib.properties:27]: and maybe cftlib gets the yaml treatment as well?

File: src/test/contractTest/resources/application.properties [hunk 3]
```diff
@@ -19,12 +19,12 @@ feign.httpclient.enabled = true
 tornado.url=${TORNADO_URL:http://localhost:8090/rs/render}
 tornado.accessKey=${TORNADO_ACCESS_KEY:}
 
-ccd.data.store.api.url=${CCD_DATA_STORE_API_URL:http://ccd-data-store-api:4452}
+ccd.data-store-api-url=${CCD_DATA_STORE_API_URL:http://ccd-data-store-api:4452}
```
Comment (hsjhita) [inline src/test/contractTest/resources/application.properties:22]: maybe all properties files should be refactored into yaml files just for consistency?

File: src/main/resources/application-cftlib.properties (hunk 2 above)
Comment (JackReevies) [reply src/main/resources/application-cftlib.properties:27]: Good spot, yes, can delete all these. I'll upgrade to yaml too

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/DocumentManagementService.java (hunk 1 above)
Comment (JackReevies) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/DocumentManagementService.java:208]: Yeah the consumer has changed behaviour so this needs to change to reflect it, its only used in these two places

**Review by hsjhita — APPROVED** [review]

**Review by hsjhita — APPROVED** [review]

**Review by furnivall — APPROVED** [review]
Approval based on fairly cursory read through and local testing for 3 devs on PET team which appeared to work.

**Pull request merged**